frontend/dockerfile/builder/build.go

```diff
@@ -324,6 +325,12 @@ func Build(ctx context.Context, c client.Client) (*client.Result, error) {
 				return nil, wrapSource(err, sourceMap, loc)
 			}
 			return res, err
+		} else if ref, ok := opts[keySyntax]; ok {
+			res, err := forwardGateway(ctx, c, ref, ref)
```

Comment:
look at how `DetectSyntax` handles the `cmdline` parameter. Should do the same for this so they are equivalent.

Reply:
As I tested, the value of `cmdline` is the same as `ref`, Ex:
```
# syntax = docker/dockerfile:1.1-experimental
```
`cmdline` and `ref` both are `docker/dockerfile:1.1-experimental`, so I use the same variable here.

Reply:
The difference is that ref is the first argument of syntax, cmdline is all.

Reply:
ah, It's means we can add more arguments like `# syntax=docker/dockerfile:experimental  MORE ARGS` ? 
but I did not find any description in `https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/experimental.md`  😕

Reply:
I don't think it is used by anything atm but it exists for future compatibility

Reply:
ok, got it, thanks, let me update it soon.